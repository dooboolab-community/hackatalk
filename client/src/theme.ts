import type {DoobooTheme} from 'dooboo-ui';

/* eslint-disable */
export const colors = {
  brand: '#1E6EFA',
  paleBlue05: '#E2F2FF',
  paleBlue10: '#8BC9FF',
  green10: '#A1EEDB',
  green30: '#00D4AB',
  green50: '#28DB98',
  gray05: '#F3F9FF',
  success: '33ff2f',
  danger: '#ff002e',
  warning: '#f2df0f',
  info: '#3a74e7',
  light: '#ededed',
  dark: '#151C22',
  messagePrimary: '#8BC9FF',
  messageSecondary: '#E2F2FF',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: 'white',
  paper: '#F2F5F6',
  disabled: '#C4C4C4',
  line: '#gray05',
  text: '#000',
  textLight: '#1B1B1B',
  textContrast: 'white',
  placeholder: '#6D6D6D',
  header: colors.brand,
  card: 'white',
  active: colors.brand,
  primary: colors.brand,
  primaryLight: colors.paleBlue10,
  secondary: colors.green10,
  button: colors.brand,
};

export type Theme = typeof light & DoobooTheme;

export const dark = {
  background: '#000',
  paper: colors.dark,
  disabled: '#515151',
  line: '#000',
  text: 'white',
  textLight: '#D3D3D3',
  textContrast: '#000',
  placeholder: '#6D6D6D',
  header: '#000',
  card: colors.dark,
  active: colors.paleBlue10,
  primary: '#fff',
  primaryLight: '#151A25',
  secondary: colors.green30,
  button: colors.paleBlue10,
};
