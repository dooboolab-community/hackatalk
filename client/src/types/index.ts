import { StyleProp, TextStyle } from 'react-native';

import { SFC } from 'react';
import { User } from './graphql';

export enum MessageType {
  Message,
  Photo,
  File,
}

export interface Friend extends User {
  isFriend?: boolean;
  friendSince: Date;
}

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
