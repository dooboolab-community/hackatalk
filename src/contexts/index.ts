import React from 'react';

export const AppContext = React.createContext<any>(null);
