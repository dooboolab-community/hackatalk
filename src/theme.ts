/* eslint-disable */
export const colors = {
  light: '#ffffff',
  dark: '#151A25',
  backgroundDark: '#151E22',
  deepDark: '#000000',
  blue_80: '#2F38C7',
  blue_70: '#1E6EFA',
  blue_60: '#217DFE',
  paleBlue_10: '#8bc9ff',
  paleBlue_05: '#e2f2ff',
  green_30: '#00D4AB',
  green_10: '#A1EEDB',
  gray_80: '#676c7a',
  gray_70: '#545966',
  gray_60: '#676c7a',
  gray_50: '#8f94a3',
  gray_40: '#AFB4C3',
  gray_30: '#d3d8e8',
  gray_05: '#f3f9ff',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: colors.light,
  backgroundDark: colors.paleBlue_05,
  itemBackground: colors.light,
  btnPrimary: colors.blue_70,
  btnPrimaryFont: colors.light,
  btnPrimaryLight: colors.light,
  btnPrimaryLightFont: colors.blue_70,
  textDisabled: colors.gray_50,
  btnDisabled: colors.gray_30,
  fontColor: colors.dark,
  fontSubColor: colors.blue_70,
  labelColor: colors.dark,
  tintColor: colors.green_30,
  lineColor: colors.gray_30,
  indicatorColor: colors.blue_70,
  inactiveColor: colors.gray_60,
  primary: colors.blue_70,
  primaryLight: colors.paleBlue_10,
  headerFont: colors.light,
  searchBackground: colors.gray_05,
  messageBackground: colors.light,
  status: colors.green_10,
  placeholder: colors.gray_60,
  focused: colors.blue_70,
  placeholderFocused: colors.blue_70,
  socialBtn: colors.paleBlue_05,
  link: colors.blue_70,
  myMessageBackground: colors.paleBlue_10,
  myMessageText: colors.dark,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.light,
  modalBtnPrimaryFont: colors.blue_80,
  modalBtnFont: colors.gray_60,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.light,
  facebookText: colors.light,
  facebookBackground: colors.facebook,
  googleIcon: colors.light,
  googleText: colors.light,
  googleBackground: colors.google,
  errorTitle: colors.gray_60,
  errorBody: colors.gray_50,
  fab: colors.blue_70,
};
export type Theme = typeof light;
export const dark = {
  background: colors.deepDark,
  backgroundDark: colors.deepDark,
  itemBackground: colors.dark,
  btnPrimary: colors.paleBlue_10,
  btnPrimaryFont: colors.deepDark,
  btnPrimaryLight: colors.dark,
  btnPrimaryLightFont: colors.paleBlue_10,
  textDisabled: colors.gray_80,
  btnDisabled: colors.gray_40,
  fontColor: colors.light,
  fontSubColor: colors.paleBlue_10,
  labelColor: colors.light,
  tintColor: colors.green_10,
  lineColor: colors.light,
  indicatorColor: colors.paleBlue_10,
  inactiveColor: colors.gray_50,
  primary: colors.deepDark,
  primaryLight: colors.dark,
  headerFont: colors.light,
  searchBackground: colors.gray_80,
  messageBackground: colors.dark,
  status: colors.green_30,
  placeholder: colors.gray_50,
  focused: '#ffffff',
  placeholderFocused: colors.paleBlue_10,
  socialBtn: colors.gray_05,
  link: colors.paleBlue_05,
  myMessageBackground: colors.deepDark,
  myMessageText: colors.light,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.gray_80,
  modalBtnPrimaryFont: colors.paleBlue_10,
  modalBtnFont: colors.gray_40,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.facebook,
  facebookText: colors.dark,
  facebookBackground: colors.light,
  googleIcon: colors.google,
  googleText: colors.dark,
  googleBackground: colors.light,
  errorTitle: colors.paleBlue_10,
  errorBody: colors.gray_40,
  fab: colors.paleBlue_10,
};
