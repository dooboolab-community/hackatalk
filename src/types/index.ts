import { StyleProp, TextStyle } from 'react-native';

import { SFC } from 'react';

export enum MessageType {
  Message,
  Photo,
  File,
}

export enum AuthType {
  EMAIL,
  FACEBOOK,
  GOOGLE,
  APPLE
}

export interface User {
  id: string;
  email?: string;
  nickname?: string;
  birthday?: Date;
  name?: string;
  thumbURL?: string;
  photoURL?: string;
  statusMessage?: string;
  isOnline?: boolean;
  socialId?: string;
  lastSignedIn?: string;
  authType?: AuthType;
  verified?: boolean;
  created?: string;
  updated?: string;
}

export interface AuthPayload {
  token: string;
  user: User;
}

export interface Friend extends User {
  isFriend?: boolean;
  friendSince: Date;
}

export interface MessageCommon<T extends MessageType = MessageType.Message> {
  id: string;
  sender: User;
  messageType?: T;
  created?: string;
  updated?: string;
}

interface Message extends MessageCommon<MessageType.Message> {
  message: string;
}

interface Photo extends MessageCommon<MessageType.Photo> {
  photo: string;
}

interface File extends MessageCommon<MessageType.File> {
  file: string;
}

export type MessageProps = Message | Photo | File;

export interface ChannelType {
  id: string;
  lastMessage: MessageProps;
  lastMessageCnt: number;
  messages?: MessageProps[];
  users?: User[];
  created?: Date;
  updated?: Date;
}

// export type Friend = Omit<User, 'uid'>;

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
