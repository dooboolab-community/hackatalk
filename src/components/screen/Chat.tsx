import { Image, Platform, TouchableOpacity, View } from 'react-native';
import {
  NavigationParams,
  NavigationScreenProp,
  NavigationState,
} from 'react-navigation';
import React, { useState } from 'react';
import styled, {
  DefaultTheme,
  ThemeProps,
  withTheme,
} from 'styled-components/native';

import Button from '../shared/Button';
import { Chat } from '../../types';
import ChatListItem from '../shared/ChatListItem';
import Constants from 'expo-constants';
import EmptyListItem from '../shared/EmptyListItem';
import GiftedChat from '../shared/GiftedChat';
import { Header } from 'react-navigation-stack';
import { IC_SMILE } from '../../utils/Icons';
import { Ionicons } from '@expo/vector-icons';
import { getString } from '../../../STRINGS';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: column;
  align-items: center;
`;

interface Props extends ThemeProps<DefaultTheme> {
  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
}

interface State {
  isLoading: boolean;
  showMenu: boolean;
  message: string;
  chats: Chat[];
}

function Screen(props: Props): React.ReactElement {
  const { theme } = props;

  const [isSending, setIsSending] = useState<boolean>(false);
  const [message, setMessage] = useState<string>('');
  const [chats, setChats] = useState<Chat[]>([
    {
      id: '',
      sender: {
        uid: '0',
        displayName: 'sender111',
        thumbURL: '',
        photoURL: '',
        statusMsg: '',
      },
      message: 'hello1',
    },
    {
      id: '',
      sender: {
        uid: '2',
        displayName: 'sender111',
        thumbURL: '',
        photoURL: '',
        statusMsg: '',
      },
      message:
        'Hello2. This is long message. This is long message.This is long message.' +
        'This is long message. This is long message. This is long message.' +
        'This is long message. This is long message.' +
        'This is long message. This is long message. This is long message.',
    },
    {
      id: '',
      sender: {
        uid: '0',
        displayName: 'sender111',
        thumbURL: '',
        photoURL: '',
        statusMsg: '',
      },
      message: 'hello',
    },
    {
      id: '',
      sender: {
        uid: '0',
        displayName: 'sender111',
        thumbURL: '',
        photoURL: '',
        statusMsg: '',
      },
      message: 'hello2',
    },
  ]);

  const onSubmit = (): void => {
    setIsSending(true);
  };

  return (
    <StyledContainer>
      <GiftedChat
        chats={chats}
        borderColor={theme.lineColor}
        backgroundColor={theme.background}
        fontColor={theme.fontColor}
        keyboardOffset={Constants.statusBarHeight + Header.HEIGHT}
        message={message}
        placeholder={getString('WRITE_MESSAGE')}
        placeholderTextColor={theme.status}
        onChangeMessage={(text: string): void => setMessage(text)}
        renderItem={({
          item,
          index,
        }: {
          item: Chat;
          index: number;
        }): React.ReactElement => {
          return (
            <ChatListItem
              prevItem={index > 0 ? chats[index - 1] : undefined}
              item={item}
            />
          );
        }}
        optionView={
          <Image
            style={{
              width: 20,
              height: 20,
            }}
            source={IC_SMILE}
          />
        }
        emptyItem={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        renderViewMenu={(): React.ReactElement => (
          <View
            style={{
              flexDirection: 'row',
              marginTop: 10,
            }}
          >
            <TouchableOpacity
              style={{
                marginLeft: 16,
                marginTop: 2,
                width: 60,
                height: 60,
                justifyContent: 'center',
                alignItems: 'center',
              }}
            >
              <Ionicons
                name="ios-camera"
                size={36}
                color={theme ? theme.fontColor : '#3d3d3d'}
              />
            </TouchableOpacity>
            <TouchableOpacity
              style={{
                marginLeft: 16,
                marginTop: 4,
                width: 60,
                height: 60,
                justifyContent: 'center',
                alignItems: 'center',
              }}
            >
              <Ionicons
                name="md-images"
                size={36}
                color={theme ? theme.fontColor : '#3d3d3d'}
              />
            </TouchableOpacity>
          </View>
        )}
        renderSendButton={(): React.ReactElement => (
          <Button
            testID="btn_chat"
            height={Platform.OS === 'android' ? 40 : undefined}
            isLoading={isSending}
            onPress={onSubmit}
          >
            {getString('SEND')}
          </Button>
        )}
      />
    </StyledContainer>
  );
}

export default withTheme(Screen);
