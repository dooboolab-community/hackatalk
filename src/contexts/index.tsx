import {
  ProfileModalAction,
  ProfileModalState,
  initialProfileModal,
  profileModalReducer,
} from '../reducers/ProfileModalReducer';
import React, {
  Dispatch,
  FC,
  ReactNode,
  createContext,
  useContext,
  useReducer,
  useRef,
} from 'react';
import {
  ThemeAction,
  initialTheme,
  themeReducer,
} from '../reducers/themeReducer';

import { ThemeType } from '../types';

// TODO: preferred to invent a way to automatically overload or joined typescript perk
interface MainState {
  theme: ThemeType;
  profileModal: ProfileModalState;
}

type TMainAction = ThemeAction | ProfileModalAction;

const mainInitialState = {
  theme: initialTheme,
  profileModal: initialProfileModal,
};

const mainReducer = (
  { theme, profileModal }: MainState,
  action: TMainAction,
): MainState => {
  // middleware goes here, i.e. calling analytics service, etc.
  return {
    theme: themeReducer(theme, action),
    profileModal: profileModalReducer(profileModal, action),
  };
};

export const StateContext = createContext<[MainState, Dispatch<TMainAction>]>([
  mainInitialState,
  (): null => null,
]);

export const StateProvider: FC<ReactNode> = ({ children }) => {
  const modal = useRef<any>(null);

  const modalInit = (): MainState => {
    return {
      ...mainInitialState,
      profileModal: {
        ...mainInitialState.profileModal,
        modal,
      },
    };
  };
  return (
    <StateContext.Provider
      value={useReducer(mainReducer, mainInitialState, modalInit)}
    >
      {children}
    </StateContext.Provider>
  );
};

export const useStateValue = (): [MainState, Dispatch<TMainAction>] =>
  useContext(StateContext);
