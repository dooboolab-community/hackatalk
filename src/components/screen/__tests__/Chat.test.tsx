import 'react-native';

import React, { ReactElement } from 'react';
import {
  RenderResult,
  act,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react-native';

import Chat from '../Chat';
import { StateProvider } from '../../../contexts';
import { ThemeProvider } from 'styled-components/native';
import { ThemeType } from '../../../types';
import { createTheme } from '../../../theme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const props = {
  navigation: {
    navigate: jest.fn(),
  },
  createTheme,
};

const component: ReactElement = (
  <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
    <Chat {...props} />
  </ThemeProvider>
);

describe('[Chat] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[Login] interaction', () => {
  let testingLib: RenderResult;

  beforeAll(() => {
    testingLib = render(component);
  });

  it('should invoke changeText event handler when message changed', () => {
    const textInput = testingLib.getByTestId('input_chat');
    jest.useFakeTimers();
    jest.runAllTimers();
    fireEvent.changeText(textInput, 'chat test');
    expect(textInput.props.value).toEqual('chat test');
  });

  it('should call [setShowMenu] when focused', () => {
    const textInput = testingLib.getByTestId('input_chat');
    textInput.props.onFocus();
  });

  it('should [showMenu] when touch pressed', () => {
    let touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent.press(touchMenu);

    touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent.press(touchMenu);
  });

  it('should call [setShowMenu] when focused', () => {
    const touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent.press(touchMenu);
  });

  it('should [sendChat] when pressing button', () => {
    let chatBtn = testingLib.getByTestId('btn_chat');
    fireEvent.press(chatBtn);

    const touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent.press(touchMenu);

    chatBtn = testingLib.getByTestId('btn_chat');
    fireEvent.press(chatBtn);
  });
});
