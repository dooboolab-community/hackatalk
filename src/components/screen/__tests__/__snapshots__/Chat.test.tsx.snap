// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Chat] rendering test renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Shared
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          No content
        </Shared>
      }
      contentContainerStyle={null}
      data={
        Array [
          Object {
            "id": "",
            "message": "hello1",
            "sender": Object {
              "displayName": "sender111",
              "photoURL": "",
              "statusMsg": "",
              "thumbURL": "",
              "uid": "0",
            },
          },
          Object {
            "id": "",
            "message": "Hello2. This is long message. This is long message.This is long message.This is long message. This is long message. This is long message.This is long message. This is long message.This is long message. This is long message. This is long message.",
            "sender": Object {
              "displayName": "sender111",
              "photoURL": "",
              "statusMsg": "",
              "thumbURL": "",
              "uid": "2",
            },
          },
          Object {
            "id": "",
            "message": "hello",
            "sender": Object {
              "displayName": "sender111",
              "photoURL": "",
              "statusMsg": "",
              "thumbURL": "",
              "uid": "0",
            },
          },
          Object {
            "id": "",
            "message": "hello2",
            "sender": Object {
              "displayName": "sender111",
              "photoURL": "",
              "statusMsg": "",
              "thumbURL": "",
              "uid": "0",
            },
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "alignSelf": "stretch",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            isSame={false}
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "marginTop": 20,
                  "minHeight": 48,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                  "width": 40,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <View
                  style={
                    Object {
                      "width": 40,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "maxWidth": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "marginBottom": 2,
                    },
                  ]
                }
              >
                sender111
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "rgb(221,226,236)",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "marginRight": 8,
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  hello1
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "marginRight": 20,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                  "marginTop": 20,
                  "minHeight": 48,
                  "width": "100%",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282027007,
                    4284794879,
                  ],
                  "endPoint": Array [
                    1,
                    0.6,
                  ],
                  "locations": Array [
                    0,
                    1,
                  ],
                  "startPoint": Array [
                    0.7,
                    0.4,
                  ],
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "marginLeft": 28,
                    "marginRight": 20,
                    "paddingBottom": 12,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 12,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Hello2. This is long message. This is long message.This is long message.This is long message. This is long message. This is long message.This is long message. This is long message.This is long message. This is long message. This is long message.
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                Array [
                  Object {
                    "color": "#50e3c2",
                    "fontSize": 12,
                    "marginRight": 20,
                    "marginTop": 4,
                  },
                ]
              }
            >
              0 : 0
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            isSame={false}
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "marginTop": 20,
                  "minHeight": 48,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                  "width": 40,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <View
                  style={
                    Object {
                      "width": 40,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "maxWidth": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "marginBottom": 2,
                    },
                  ]
                }
              >
                sender111
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "rgb(221,226,236)",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "marginRight": 8,
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  hello
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "marginRight": 20,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            isSame={true}
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "marginTop": 8,
                  "minHeight": 48,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                  "width": 40,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <View
                  style={
                    Object {
                      "width": 40,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "maxWidth": "90%",
                }
              }
            >
              <View />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "rgb(221,226,236)",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "marginRight": 8,
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  hello2
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 12,
                      "marginRight": 20,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(221,226,236)",
            "borderTopWidth": 0.5,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
            "maxHeight": 52,
            "minHeight": 52,
            "paddingLeft": 8,
            "paddingRight": 8,
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        multiline={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Write message..."
        placeholderTextColor="#50e3c2"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "color": "black",
              "fontSize": 14,
              "marginRight": 20,
              "paddingLeft": 48,
              "width": "80%",
            },
          ]
        }
        testID="input_chat"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "left": 10,
            "minWidth": 20,
            "opacity": 1,
            "position": "absolute",
          }
        }
        testID="touch_menu"
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginVertical": 8,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="btn_chat"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(58,139,255)",
                  "borderColor": "white",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "justifyContent": "center",
                },
              ]
            }
            white={false}
          >
            <View
              height={60}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 60,
                    "justifyContent": "center",
                    "width": 136,
                  },
                ]
              }
              width={136}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                  ]
                }
                white={false}
              >
                Send
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
